src/functionalTest/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/OcrFormValidationTest.java

```diff
@@ -0,0 +1,130 @@
+        this.s2sUrl = config.getString("test-s2s-url");
+        this.s2sName = config.getString("test-s2s-name");
+        this.s2sSecret = config.getString("test-s2s-secret");
+    }
+
+    @Test
+    void should_validate_ocr_data_and_return_success() {
+        Response response = sendOcrFormValidationRequest("PERSONAL", "valid-ocr-form-data.json");
+
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        OcrValidationResponse validationResponse = response.getBody()
+            .as(OcrValidationResponse.class, ObjectMapperType.JACKSON_2);
+
+        OcrValidationResponse expectedResponse = new OcrValidationResponse(
+            emptyList(), emptyList(), ValidationStatus.SUCCESS
+        );
+
+        assertThat(validationResponse).isEqualToComparingFieldByFieldRecursively(expectedResponse);
+    }
+
+    @Test
+    void should_return_errors_when_mandatory_fields_are_missing() {
+        Response response = sendOcrFormValidationRequest("PERSONAL", "missing-mandatory-fields.json");
+
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        OcrValidationResponse validationResponse = response.getBody()
+            .as(OcrValidationResponse.class, ObjectMapperType.JACKSON_2);
+
+        OcrValidationResponse expectedResponse = new OcrValidationResponse(
+            emptyList(),
+            singletonList("first_name is missing"),
+            ValidationStatus.ERRORS
+        );
+
+        assertThat(validationResponse).isEqualToComparingFieldByFieldRecursively(expectedResponse);
+    }
+
+    @Test
+    void should_return_errors_when_optional_fields_are_missing() {
+        Response response = sendOcrFormValidationRequest("CONTACT", "missing-optional-fields.json");
+
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        OcrValidationResponse validationResponse = response.getBody()
+            .as(OcrValidationResponse.class, ObjectMapperType.JACKSON_2);
+
+        OcrValidationResponse expectedResponse = new OcrValidationResponse(
+            asList("post_tows is missing", "county is missing"),
+            emptyList(),
+            ValidationStatus.WARNINGS
+        );
+
+        assertThat(validationResponse).isEqualToComparingFieldByFieldRecursively(expectedResponse);
+    }
+
+    @Test
+    void should_return_errors_when_additional_validations_are_failing() {
+        Response response = sendOcrFormValidationRequest("CONTACT", "invalid-form-data.json");
+
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        OcrValidationResponse validationResponse = response.getBody()
+            .as(OcrValidationResponse.class, ObjectMapperType.JACKSON_2);
+
+        OcrValidationResponse expectedResponse = new OcrValidationResponse(
+            emptyList(),
+            asList("Invalid email address", "Invalid phone number"),
+            ValidationStatus.ERRORS
+        );
+
+        assertThat(validationResponse).isEqualToComparingFieldByFieldRecursively(expectedResponse);
```

Comment:
maybe instead of 7 lines above we could simply do:
```
assertThat(validationResponse.status).isEqualTo(ERRORS);
assertThat(validationResponse.errors).containsExactly("Invalid email address", "Invalid phone number");
assertThat(validationResponse.warnings).isEmpty();
```
Up to you.

Imho clearer to see what's what in code and also easier to understand potential failed assertion messages.

Reply:
Sure, will change